### PR TITLE
Update transmission parser to parse the joint role

### LINKED ISSUE #317: Joint role tag is not parsed from urdf but it exists in the JointInfo struct
The `transmission_interface::JointInfo` struct has `role_` as a data member but the transmission parser doesn't parse it.

### PR DESCRIPTION
**Pull request by jlack1987**
This is to fix #317

### CODE REVIEW

**Comment by jlack1987** [comment]
No clue how my commits caused a CI failure. The following looks like the error output that's causing the problem,
```
Full test results for 'build/transmission_interface/test_results/transmission_interface/MISSING-gtest-differential_transmission_loader_test.xml'
-------------------------------------------------
<?xml version="1.0" encoding="UTF-8"?>
<testsuite tests="1" failures="1" time="1" errors="0" name="gtest-differential_transmission_loader_test.xml">
  <testcase name="test_ran" status="run" time="1" classname="Results">
    <failure message="Unable to find test results for gtest-differential_transmission_loader_test.xml, test did not run.
Expected results in /root/catkin_ws/build/transmission_interface/test_results/transmission_interface/gtest-differential_transmission_loader_test.xml" type=""/>
  </testcase>
</testsuite>
-------------------------------------------------
Full test results for 'build/transmission_interface/test_results/transmission_interface/MISSING-gtest-four_bar_linkage_transmission_loader_test.xml'
-------------------------------------------------
<?xml version="1.0" encoding="UTF-8"?>
<testsuite tests="1" failures="1" time="1" errors="0" name="gtest-four_bar_linkage_transmission_loader_test.xml">
  <testcase name="test_ran" status="run" time="1" classname="Results">
    <failure message="Unable to find test results for gtest-four_bar_linkage_transmission_loader_test.xml, test did not run.
Expected results in /root/catkin_ws/build/transmission_interface/test_results/transmission_interface/gtest-four_bar_linkage_transmission_loader_test.xml" type=""/>
  </testcase>
</testsuite>
-------------------------------------------------
build/transmission_interface/test_results/transmission_interface/MISSING-gtest-differential_transmission_loader_test.xml: 1 tests, 0 errors, 1 failures, 0 skipped
build/transmission_interface/test_results/transmission_interface/MISSING-gtest-four_bar_linkage_transmission_loader_test.xml: 1 tests, 0 errors, 1 failures, 0 skipped
```
not sure how my commit had an affect on that

**Comment by bmagyar** [comment]
I've ran the tests in a local docker container. Note below the **Segmentation fault (core dumped)** appearing on both tests that report failure here.

```
-- run_tests.py: execute commands
  /root/ws/devel/lib/transmission_interface/four_bar_linkage_transmission_loader_test --gtest_output=xml:/root/ws/build/test_results/transmission_interface/gtest-four_bar_linkage_transmission_loader_test.xml
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from FourBarLinkageTransmissionLoaderTest
[ RUN      ] FourBarLinkageTransmissionLoaderTest.FullSpec
[       OK ] FourBarLinkageTransmissionLoaderTest.FullSpec (20 ms)
[ RUN      ] FourBarLinkageTransmissionLoaderTest.MinimalSpec
[       OK ] FourBarLinkageTransmissionLoaderTest.MinimalSpec (1 ms)
[ RUN      ] FourBarLinkageTransmissionLoaderTest.InvalidSpec
Segmentation fault (core dumped)
-- run_tests.py: verify result "/root/ws/build/test_results/transmission_interface/gtest-four_bar_linkage_transmission_loader_test.xml"
Cannot find results, writing failure results to '/root/ws/build/test_results/transmission_interface/MISSING-gtest-four_bar_linkage_transmission_loader_test.xml'
[ 56%] Built target run_tests_transmission_interface_gtest_four_bar_linkage_transmission_loader_test
[ 68%] Built target differential_transmission_loader_test
Scanning dependencies of target run_tests_transmission_interface_gtest_differential_transmission_loader_test
-- run_tests.py: execute commands
  /root/ws/devel/lib/transmission_interface/differential_transmission_loader_test --gtest_output=xml:/root/ws/build/test_results/transmission_interface/gtest-differential_transmission_loader_test.xml
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from DifferentialTransmissionLoaderTest
[ RUN      ] DifferentialTransmissionLoaderTest.FullSpec
[       OK ] DifferentialTransmissionLoaderTest.FullSpec (26 ms)
[ RUN      ] DifferentialTransmissionLoaderTest.MinimalSpec
[       OK ] DifferentialTransmissionLoaderTest.MinimalSpec (1 ms)
[ RUN      ] DifferentialTransmissionLoaderTest.InvalidSpec
Segmentation fault (core dumped)
-- run_tests.py: verify result "/root/ws/build/test_results/transmission_interface/gtest-differential_transmission_loader_test.xml"
```

**Comment by jlack1987** [comment]
It looks like the problem is the `GetText()` function call returns a `nullptr` if the field is an empty string which is what's happening. I fixed it, but noticed there's lots of other places in the transmission parser code where this could result in a segfault. Found [this thread](https://github.com/leethomason/tinyxml2/issues/105) where it talks about it a bit.

**Comment by bmagyar** [comment]
I see. Not sure if this is better, it's definitely more brief:
```
joint.role_ =  role_it->GetText() ? role_it->GetText() : std::string()
```

**Comment by jlack1987** [comment]
Agreed it is more concise, thanks for the suggestion. Have made the update.

File: transmission_interface/src/transmission_parser.cpp [hunk 1]
```diff
@@ -146,6 +146,12 @@ bool TransmissionParser::parseJoints(TiXmlElement *trans_it, std::vector<JointIn
       return false;
     }
 
+    TiXmlElement *role_it = joint_it->FirstChildElement("role");
+    if(role_it)
+    {
+      joint.role_ =  role_it->GetText() ? role_it->GetText() : std::string();
```
Comment (bmagyar) [inline transmission_interface/src/transmission_parser.cpp]: There's an extra space after "="

**Review by bmagyar — CHANGES_REQUESTED** [review]
Could you please add a test that ensures that an empty role tag gets us an empty string?

File: transmission_interface/src/transmission_parser.cpp (hunk 1 above)
Comment (jlack1987) [reply transmission_interface/src/transmission_parser.cpp]: Fixed

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
Thanks!
